sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/convert/MappingCosmosConverter.java

```diff
@@ -96,6 +99,27 @@ public void write(Object source, JsonNode sink) {
         }
     }
 
+    private <R> R readJsonNodeAsValue(JsonNode jsonNode, Class<R> type) {
+        switch (jsonNode.getNodeType()) {
+            case BOOLEAN:
```

Comment:
Shouldn't we special case ARRAY here as well?

Reply:
@FabianMeiswinkel - updated the logic to convert value nodes, objectMapper should automatically take care of that. Will test it out as well to verify.